index.php

```diff
@@ -179,81 +292,19 @@ function verify_password($url, $pass)
-        <input type="hidden" name="client_id" value="<?php echo htmlspecialchars($client_id)?>" />
-        <div class="form-line"><label for="password">Password:</label> <input type="password" name="password" id="password" /></div>
-        <div class="form-line"><input class="submit" type="submit" name="submit" value="Submit" /></div>
-    </form>
-
-    </body></html>
-<?php 
-    exit();
-} //end elseif
-
-$csrf_code      = (isset($_POST['_csrf']        ) ? $_POST['_csrf']           : null );  
-$pass_input     = (isset($_POST['password']     ) ? $_POST['password']        : null );  
-$me             = (isset($_POST['me']           ) ? $_POST['me']              : null );  
-$redirect_uri   = (isset($_POST['redirect_uri'] ) ? $_POST['redirect_uri']    : null );
-$response_type  = (isset($_POST['response_type']) ? $_POST['response_type']   : 'id' );
-$state          = (isset($_POST['state']        ) ? $_POST['state']           : ''   );
-$client_id      = (isset($_POST['client_id']    ) ? $_POST['client_id']       : null );
-$scope          = (isset($_POST['scope']        ) ? $_POST['scope']           : '' );
-
-//TODO check scope and response_type make sense once they are supported
-
-if(!verify_signed_code(APP_KEY, $client_id . $redirect_uri . $state, $csrf_code)){
-    error_page('Invalid csrf code','Usually this means you took too long to log in. Please try again.');
-}
-if(empty($me)){
-    error_page('Incomplete Request', 'There was an error with the request.  No "me" field given.');
-}
-if(empty($client_id)){
-    error_page('Incomplete Request', 'There was an error with the request.  No "client_id" field given.');
-}
-if(empty($redirect_uri)){
-    error_page('Incomplete Request', 'There was an error with the request.  No "redirect_uri" field given.');
-}
-if(empty($state)){
-    error_page('Incomplete Request', 'There was an error with the request.  No "state" field given.');
-}
-if(empty($pass_input)){
-    error_page('Incomplete Request', 'No Password Given.');
-}
-
-// verify login
-if(verify_password($me, $pass_input)) {
-    $scope_encoded = preg_replace('/ +/', ',', trim($scope));
-
-    $code = create_signed_code(APP_KEY, USER_URL . $redirect_uri . $client_id, 5 * 60, $scope_encoded);
-
-    $final_redir = $redirect_uri;
-    if(strpos($redirect_uri, '?') === FALSE){
-        $final_redir .= '?';
-    } else {
-        $final_redir .= '&';
-    }
-    $final_redir .= http_build_query(array(
-        'code' => $code,
-        'state' => $state,
-        'me' => $me
-    ));
-
-    // redirect back
-    header('Location: ' . $final_redir);
-    exit();
-} else {
-    error_page('Login Failed', 'Invalid username or password.');
-}
-
-
+        <h1>Authenticate</h1>
+        <div>You are attempting to login with client <pre><?php echo $client_id; ?></pre></div>
+        <div>It is requesting the following scopes <pre><?php echo htmlspecialchars($scope); ?></pre></div>
```

Comment:
Since `$scope` can be `null` and it is properly checked now, we can do:
```html
<div>You are attempting to login with client <pre><?php echo $client_id; ?></pre></div>
<?php if ($scope) : ?>
<div>It is requesting the following scopes <pre><?php echo htmlspecialchars($scope); ?></pre></div>
<?php endif; ?>
<div>After login you will be redirected to  <pre><?php echo $redirect_uri; ?></pre></div>
```